lib/events/filelog.go

```diff
@@ -115,6 +115,9 @@ type FileLog struct {
 
 // EmitAuditEvent adds a new event to the log.
 func (l *FileLog) EmitAuditEvent(ctx context.Context, event AuditEvent) error {
+	l.mu.Lock()
```

Comment:
FYI Forrest is gonna try to improve this a bit so we don't hold locks over entire functions: https://github.com/gravitational/teleport/pull/6178#discussion_r602610271.

Reply:
Okay, will keep this until he improved then bring in his commit. If I don't include his fix, tests panic because other issue is raised.

Reply:
Updated.